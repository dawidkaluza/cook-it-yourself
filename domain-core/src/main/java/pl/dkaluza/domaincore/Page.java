package pl.dkaluza.domaincore;

import java.util.List;

import static pl.dkaluza.domaincore.Validator.validator;

public class Page<T> {
    private final List<T> items;
    private final int pageNumber;
    private final int totalPages;

    private Page(List<T> items, int pageNumber, int totalPages) {
        this.items = items;
        this.pageNumber = pageNumber;
        this.totalPages = totalPages;
    }

    public static <T> Factory<Page<T>> of(List<T> items) {
        return new PageFactory<>(items, 1, 1);
    }

    public static <T> Factory<Page<T>> of(List<T> items, int pageNumber, int totalPages) {
        return new PageFactory<>(items, pageNumber, totalPages);
    }

    public List<T> getItems() {
        return items;
    }

    public int getPageNumber() {
        return pageNumber;
    }

    public int getTotalPages() {
        return totalPages;
    }

    private static class PageFactory<T> extends DefaultFactory<Page<T>> {
        private PageFactory(List<T> items, int pageNumber, int totalPages) {
            super(
                ValidationExecutor.builder()
                    .withValidator(validator(items != null, "items", "Items must not be null"))
                    .withValidator(validator(pageNumber > 0, "pageNumber", "Page number must be a positive number"))
                    .withValidator(validator(totalPages > 0, "totalPages", "Total pages must be a positive number"))
                    .build(),
                () -> new Page<>(items, pageNumber, totalPages)
            );
        }
    }
}
