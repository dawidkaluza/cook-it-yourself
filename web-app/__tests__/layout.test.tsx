import {describe, expect, test, vi} from "vitest";
import {render, screen} from "@testing-library/react";
import RootLayout from "@/app/layout";
import Page from "@/app/page";

vi.mock("@/app/_api/actions", async (importOriginal) => {
  const mod = await importOriginal<typeof import("@/app/_api/actions")>();

  return {
    ...mod,
    isSignedIn: () => false,
  };
});

describe("layout component", () => {
  test("renders", () => {
    // Given, when
    const layout = render(<RootLayout><Page /></RootLayout>);

    // Then
    const navBar = screen.queryByRole("navigation");
    expect(navBar).not.toBeNull();

    const main = screen.queryByRole("main");
    expect(main).not.toBeNull();

    expect(layout.container).toMatchSnapshot();
    layout.unmount();
  });
});