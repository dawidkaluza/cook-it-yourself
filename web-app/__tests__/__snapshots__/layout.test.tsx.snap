// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`layout component > renders 1`] = `
<div>
  <html
    data-bs-theme="dark"
    lang="en"
  >
    <body>
      <nav
        class="navbar navbar-expand bg-body-tertiary"
      >
        <div
          class="container-fluid text-center"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            Cook it yourself
          </a>
          <a
            class="btn btn-primary btn-sm"
            href="/sign-in"
          >
            Sign in
          </a>
        </div>
      </nav>
      <main
        style="margin: 1em;"
      >
        <p
          class="text-center"
        >
          Welcome my cook! Use navigation above to start your cooking journey.
        </p>
      </main>
    </body>
  </html>
</div>
`;
